src/App.tsx

```diff
@@ -42,54 +43,103 @@ export function getReorderedGoods(
 }
 
 // DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+type Props = {};
+
+export class App extends React.Component<Props, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleReverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  handleSortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPABET });
+  };
+
+  handleSortByLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
 
-      <ul>
+  handleReset = () => {
+    this.setState({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
+    goods.sort((firstGood, secondGood) => {
+      switch (sortType) {
+        case SortType.ALPABET:
+          return firstGood.localeCompare(secondGood);
+        case SortType.LENGTH:
+          return firstGood.length - secondGood.length;
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      goods.reverse();
+    }
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info',
+              { 'is-light': sortType !== SortType.ALPABET })}
```

Comment:
Do the same changes all over the code.
```suggestion
            className={classNames(
              'button is-info',
              { 'is-light': sortType !== SortType.ALPABET }
            )}
```